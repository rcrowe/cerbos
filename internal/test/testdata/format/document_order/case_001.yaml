---
# - `derivedRoles` field order
description: |-
  Common dynamic roles used within the Apatr app
apiVersion: "api.cerbos.dev/v1"
derivedRoles:
  variables:
    local: 
      flagged_resource: request.resource.attr.flagged
    import: 
      - apatr_common_variables
  name: apatr_common_roles 
  definitions:
    - name: owner 
      condition: 
        match:
          expr: request.resource.attr.owner == request.principal.id
      parentRoles: ["user"] 

    - parentRoles: ["moderator"]
      name: abuse_moderator
      condition:
        match:
          expr: V.flagged_resource == true
