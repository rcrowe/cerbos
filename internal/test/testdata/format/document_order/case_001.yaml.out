---
# - `derivedRoles` field order
# - CEL formatted variables & expressions
apiVersion: "api.cerbos.dev/v1"
description: |-
  Common dynamic roles used within the Apatr app
derivedRoles:
  name: apatr_common_roles
  variables:
    import:
      - apatr_common_variables
    local:
      flagged_resource: request.resource.attr.flagged
  definitions:
    - name: owner
      parentRoles: ["user"]
      condition:
        match:
          expr: request.resource.attr.owner == request.principal.id
    - name: abuse_moderator
      parentRoles: ["moderator"]
      condition:
        match:
          expr: V.flagged_resource == true
